src/lxc/attach.c

```diff
@@ -897,6 +890,13 @@ static int attach_child_main(struct attach_clone_payload *payload)
 		TRACE("Loaded seccomp profile");
 	}
 
+	/* Try to set the {u,g}id combination. */
+	if (new_uid != 0 || new_gid != 0 || options->namespaces & CLONE_NEWUSER) {
+		ret = lxc_switch_uid_gid(new_uid, new_gid);
+		if (ret < 0)
+			goto on_error;
+	}
```

Comment:
Ok, that needs to be done differently but I can do it. I'll just put on top of your branch.